### PR TITLE
Adding GlobalAveragePooling1D

### PR DESCRIPTION
**Pull request by Shashi456**
So I'm relatively new to the swift ecosystem, let me know if there are any errors. Afaik my understanding of GlobalAveragePooling in a single dimension is returning the mean * the no of batches if it's a multi-batch input.

### CODE REVIEW

File: Sources/DeepLearning/Layer.swift [hunk 1]
```diff
@@ -1009,12 +1009,32 @@ public struct AvgPool2D<Scalar: TensorFlowFloatingPoint>: Layer {
...
+    /// Returns the output obtained from applying the layer to the given input.
+    ///
+    /// - Parameters:
+    ///   - input: The input to the layer.
+    ///   - context: The contextual information for the layer application, e.g. the current learning
+    ///     phase.
+    /// - Returns: The output 
+    @differentiable
+    public func applied(to input: Tensor<Scalar>, in _: Context) -> Tensor<Scalar> {
+        return input.mean(alongAxes: axis)
```
Comment (tanmayb123) [inline Sources/DeepLearning/Layer.swift]:
```suggestion
        return input.mean(alongAxes: 1).reshaped(to: [input.shape[0], input.shape[2]])
```

Comment (tanmayb123) [inline Sources/DeepLearning/Layer.swift]:
```suggestion
        return input.mean(alongAxes: 1).reshaped(to: [input.shape[0], input.shape[2]])
```

Comment (tanmayb123) [reply Sources/DeepLearning/Layer.swift]: @Shashi456 the code should compile after you make this change. If the input has this shape:

[batchSize, timeSteps, channels]

The `mean(alongAxes:)` part will make it this shape:

[batchSize, 1, channels]

And then the reshape will make it this:

[batchSize, channels]

Comment (Shashi456) [reply Sources/DeepLearning/Layer.swift]: so you passed the index of time steps for the mean, which means that is where the 1D data is stored, so it computes the mean after which you have batchsize number of datapoints and hence reshaping it that way? 
Am i getting it right?

**Comment by rxwei** [comment]
Did you sign the CLA?

**Comment by Shashi456** [comment]
@rxwei yeah i have

File: Sources/DeepLearning/Layer.swift [hunk 2]
```diff
@@ -1009,12 +1009,32 @@ public struct AvgPool2D<Scalar: TensorFlowFloatingPoint>: Layer {
     }
 }
 
+
+/// A global average pooling layer for temporal data.
+@_fixed_layout
+public struct GlobalAveragePooling1D<Scalar: TensorFlowFloatingPoint>: Layer {
+    /// Creates a global average pooling layer in 1 dimension
```
Comment (rxwei) [inline Sources/DeepLearning/Layer.swift]:
```suggestion
    /// Creates a global average pooling layer in 1 dimension.
```

Please end sentences with a period.

**Pull request merged**